pyccel/codegen/printing/pycode.py

```diff
@@ -114,12 +114,41 @@ def _print_FunctionDef(self, expr):
         return code
 
     def _print_Return(self, expr):
-        return 'return {}'.format(self._print(expr.expr))
+        code = ''
+        if expr.stmt:
+            code += self._print(expr.stmt)+'\n'
+        if expr.args:
+            ret = ','.join([self._print(i) for i in expr.expr])
+            code += 'return {}'.format(ret)
+        return code
+
+    def _print_Program(self, expr):
+        body  = self._print(expr.body)
+        body = self._indent_codestring(body)
```

Comment:
can you explain what do you mean by this?

Reply:
The c code and the fortran code is also indented. Eg the fortran code is indented here:
https://github.com/pyccel/pyccel/blob/master/pyccel/codegen/printing/fcode.py#L260-261

However this function isn't called explicitly in fcode.py. I think it must be called by the sympy printer.  I suspect that the python code indentation could/should/is handled the same way.

If that is the case then `_indent_codestring` doesn't need to be called here

Reply:
Did you look into this at all?

Reply:
yes its inherit from sympy.printing.pycode

Reply:
I've just realised why this obviously needs to be here. Python has no end clause so you have to indent as you go otherwise there's no way to tell where the end is :+1: Sorry 🤦